doc/designs/ddd/ddd-01-conn-blocking.c

```diff
@@ -160,11 +160,12 @@ int main(int argc, char **argv)
 
     b = new_conn(ctx, host_port);
     if (b == NULL) {
-        fprintf(stderr, "could not create conn\n");
+        fprintf(stderr, "could not create connection\n");
         goto fail;
     }
 
-    if (tx(b, msg, sizeof(msg)) < sizeof(msg)) {
+    l = tx(b, msg, mlen);
+    if (l < mlen) {
```

Comment:
Why not `if (tx(b, msg, sizeof(msg)) < mlen) {` ?  This reduces clarity IMO.

Reply:
If you look a few lines down, then you'll see this:

```C
    for (;;) {
        l = rx(b, buf, sizeof(buf));
        if (l <= 0)
            break;
        fwrite(buf, 1, l, stdout);
}
```
I thought it made sense to use that pattern with `tx()` as well as `rx()`.